### PR TITLE
Add settings to ignore first click on unfocused window

### PR DESCRIPTION
**Pull request by felippemr**
Implements https://github.com/jwilm/alacritty/issues/1291

### CODE REVIEW

**Comment by nixpulvis** [comment]
Was there a reason to leave out the platform checks? AFAIK this only matters for macOS.

**Comment by NickeZ** [comment]
Why is this optional? Are there ever a reason to accept a click on an unfocused window?

**Comment by chrisduerr** [comment]
@NickeZ I'd say there is, at least for tiling window managers I can see this being useful.

**Comment by felippemr** [comment]
Thanks for the feedback! I will implement t today and submit to you guys. Should I go ahead and make it an OSX only feature?

**Comment by chrisduerr** [comment]
@NickeZ With focus following mouse this PR should be completely irrelevant. However I know about quite a few people that have focus on click in i3 just because they don't want random mouse movements to change the focused window. In this case it might be desired to still start a selection without having to click on the window to focus first and then start the selection.

However I completely agree with you when it comes to the additional complexity of the configuration file and codebase. I personally don't need to be able to turn this option off since I also use the option of focus following the mouse without any clicks required. However this still would be a breaking change from the current behavior without the possibility to recover the original behavior without the configuration option.

Unless there is anyone complaining in this PR already, there certainly is an option to kinda satisfy both groups though. We can remove the configuration option from this PR, merge it and just see if anyone will complain about the regression. If there is nobody speaking up I'd just go ahead and assume the target audience I'm talking about either doesn't exist or doesn't care that much.

@felippemr Unless you have a different idea I'd recommend following through with removing the configuration option and making this always take place. I have created a branch on my personal repo with the current state of this PR so no work is getting lost.

@NickeZ Thanks for chiming in, I'd say it's very likely that a large part of users will not care about being able to configure this. I haven't used WMs that use clicks to change mouse in quite a while so any other perspective is always welcome.

**Comment by felippemr** [comment]
Thanks @chrisduerr, I will make it later today.

**Comment by felippemr** [comment]
@chrisduerr just pushed

File: src/event.rs [hunk 1]
```diff
@@ -301,9 +301,11 @@ impl<N: Notify> Processor<N> {
                         processor.received_char(c);
                     },
                     MouseInput { state, button, modifiers, .. } => {
-                        *hide_cursor = false;
-                        processor.mouse_input(state, button, modifiers);
-                        processor.ctx.terminal.dirty = true;
+                        if !cfg!(target_os = "macos") || *window_is_focused {
```
Comment (chrisduerr) [inline src/event.rs]: Hey I think there has been a misunderstanding here. This should be available on all platforms, not only MacOS.

I think MacOS doesn't do anything significantly different than floating WMs on linux/windows. So consistency across platforms is probably a good idea.

Comment (felippemr) [reply src/event.rs]: Done.

Comment (felippemr) [reply src/event.rs]: Is there anything else preventing us from merging this?

**Comment by chrisduerr** [comment]
Thanks for looking into this! LGTM!

**Pull request merged**